daemon.go

```diff
@@ -356,7 +378,69 @@ func flusher(buildStarted <-chan string, buildSuccess <-chan bool) {
 	}
 }
 
+func watchDirectories(watcher *fsnotify.Watcher) {
+	for dir := range flag_directories {
+		if !*flag_recursive {
+			if err := watcher.Add(dir); err != nil {
+				log.Fatal("watcher.Add():", err)
+			}
+
+			return
+		}
+
+		err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+			if err == nil && info.IsDir() {
+				if flag_excludedDirs.Matches(path) {
+					return filepath.SkipDir
+				}
+
+				if *flag_verbose {
+					log.Printf("Watching directory '%s' for changes.\n", path)
+				}
+
+				return watcher.Add(path)
+			}
+
+			return err
+		})
+
+		if os.IsNotExist(err) {
+			log.Fatalf("-directory=%s does not exist", dir)
+		}
+
+		if err != nil {
+			log.Fatal("filepath.Walk():", err)
+		}
+
+		if err := watcher.Add(dir); err != nil {
+			log.Fatal("watcher.Add():", err)
+		}
+	}
+}
+
+func validateFlags() {
+	if len(flag_directories) == 0 {
+		flag_directories.Set(".")
+	}
+
+	if *flag_build_dir == "" {
+		if len(flag_directories) == 1 {
+			default_build_dir := flag_directories.First()
+			flag_build_dir = &default_build_dir
+		} else {
+			fmt.Fprintf(os.Stderr, "-build-dir is required when specifying multiple watch directeries.\n")
+			os.Exit(1)
```

Comment:
You might want to use `log.Fatal()` here which does both of these things in one call.

Reply:
No problem.  I cut+pasted this from somewhere else so I was just following what existied.